core/src/test/scala/org/apache/spark/scheduler/PoolSuite.scala

```diff
@@ -178,4 +180,97 @@ class PoolSuite extends SparkFunSuite with LocalSparkContext {
+    val schedulableBuilder = new FIFOSchedulableBuilder(rootPool)
+
+    val taskSetManager0 = createTaskSetManager(stageId = 0, numTasks = 1, taskScheduler)
+    val taskSetManager1 = createTaskSetManager(stageId = 1, numTasks = 1, taskScheduler)
+
+    val properties = new Properties()
+    properties.setProperty("spark.scheduler.pool", TEST_POOL)
+
+    // FIFOSchedulableBuilder just uses rootPool so even if properties are set, related pool
+    // (testPool) is not created and TaskSetManagers are added to rootPool
+    schedulableBuilder.addTaskSetManager(taskSetManager0, properties)
+    schedulableBuilder.addTaskSetManager(taskSetManager1, properties)
+
+    assert(rootPool.getSchedulableByName(TEST_POOL) == null)
+    assert(rootPool.schedulableQueue.size == 2)
+    assert(rootPool.getSchedulableByName(taskSetManager0.name) === taskSetManager0)
+    assert(rootPool.getSchedulableByName(taskSetManager1.name) === taskSetManager1)
+  }
+
+  test("SPARK-18066: FAIR Scheduler uses default pool when spark.scheduler.pool property is not " +
+    "set") {
+    sc = new SparkContext("local", "PoolSuite")
+    val taskScheduler = new TaskSchedulerImpl(sc)
+
+    val rootPool = new Pool("", SchedulingMode.FAIR, initMinShare = 0, initWeight = 0)
+    val schedulableBuilder = new FairSchedulableBuilder(rootPool, sc.conf)
+    schedulableBuilder.buildPools()
+
+    // FAIR Scheduler uses default pool when pool properties are null
+    val taskSetManager0 = createTaskSetManager(stageId = 0, numTasks = 1, taskScheduler)
+
+    schedulableBuilder.addTaskSetManager(taskSetManager0, null)
+
+    val defaultPool = rootPool.getSchedulableByName(schedulableBuilder.DEFAULT_POOL_NAME)
+    assert(defaultPool != null)
+    assert(defaultPool.schedulableQueue.size == 1)
+    assert(defaultPool.getSchedulableByName(taskSetManager0.name) === taskSetManager0)
+
+    // FAIR Scheduler uses default pool when spark.scheduler.pool property is not set
+    val taskSetManager1 = createTaskSetManager(stageId = 1, numTasks = 1, taskScheduler)
+
+    schedulableBuilder.addTaskSetManager(taskSetManager1, new Properties())
+
+    assert(defaultPool.schedulableQueue.size == 2)
+    assert(defaultPool.getSchedulableByName(taskSetManager1.name) === taskSetManager1)
+
+    // FAIR Scheduler uses default pool when spark.scheduler.pool property is set as default pool
+    val taskSetManager2 = createTaskSetManager(stageId = 2, numTasks = 1, taskScheduler)
+
+    val properties = new Properties()
+    properties.setProperty(schedulableBuilder.FAIR_SCHEDULER_PROPERTIES, schedulableBuilder
+      .DEFAULT_POOL_NAME)
+
+    schedulableBuilder.addTaskSetManager(taskSetManager2, properties)
+
+    assert(defaultPool.schedulableQueue.size == 3)
+    assert(defaultPool.getSchedulableByName(taskSetManager2.name) === taskSetManager2)
+  }
+
+  test("SPARK-18066: FAIR Scheduler creates a new pool when spark.scheduler.pool property points " +
+    "non-existent") {
```

Comment:
yeah I prefer failing fast for the same reason.  But I guess we can't actually change this, since there is a valid use case for this behavior.  We don't really know if others are using this, and I don't think we should change it under them in the next release.

In any case, I still think we should also change `Pool.weight` and `Pool.minShare` to `val`s.